subworkflows/local/align_cellrangermulti.nf

```diff
@@ -0,0 +1,139 @@
+//
+// Include modules
+//
+include { CELLRANGER_MKGTF    } from "../../modules/nf-core/cellranger/mkgtf/main.nf"
+include { CELLRANGER_MKREF    } from "../../modules/nf-core/cellranger/mkref/main.nf"
+include { CELLRANGER_MKVDJREF } from "../../modules/nf-core/cellranger/mkvdjref/main.nf"
+include { CELLRANGER_MULTI    } from "../../modules/nf-core/cellranger/multi/main.nf"
+
+// Define workflow to subset and index a genome region fasta file
+workflow CELLRANGER_MULTI_ALIGN {
+    take:
+        ch_fasta
+        ch_gtf
+        ch_fastq
+        cellranger_gex_index
+        cellranger_vdj_index
+        empty_file
+
+    main:
+        ch_versions    = Channel.empty()
+
+        // since we merged all data as a meta, now we have a channel per sample, which
+        // every item is a meta map for each data-type
+        // now we can split it back for passing as input to the module
+        ch_fastq
+        .flatten()
+        .map{ meta ->
+            def meta_clone = meta.clone()
+            def data_dict  = meta_clone.find{ it.key == "${meta_clone.feature_type}" }
+            fastqs = data_dict?.value
+            meta_clone.remove( data_dict?.key )
+            [ meta_clone, fastqs ]
+        }
+        .branch {
+            meta, fastq ->
+                gex: meta.feature_type == "gex"
+                    return [ meta, fastq ]
+                vdj: meta.feature_type == "vdj"
+                    return [ meta, fastq ]
+                ab: meta.feature_type == "ab"
+                    return [ meta, fastq ]
+                beam: meta.feature_type == "beam"
+                    return [ meta, fastq ]
+                crispr: meta.feature_type == "crispr"
+                    return [ meta, fastq ]
+                cmo: meta.feature_type == "cmo"
+                    return [ meta, fastq ]
+        }
+        .set { ch_grouped_fastq }
+
+        // Assign other cellranger reference files
+        ch_gex_frna_probeset      = params.gex_frna_probe_set ? file(params.gex_frna_probe_set) : empty_file
+        ch_gex_target_panel       = params.gex_target_panel   ? file(params.gex_target_panel)   : empty_file
+        ch_gex_cmo_set            = params.gex_cmo_set        ? file(params.gex_cmo_set)        : empty_file
+        ch_gex_barcodes           = params.gex_barcode_sample_assignment ? file(params.gex_barcode_sample_assignment) : empty_file
+        ch_fb_reference           = params.fb_reference       ? file(params.fb_reference)       : empty_file
+        ch_vdj_primer_index       = params.vdj_inner_enrichment_primers ? file(params.vdj_inner_enrichment_primers) : empty_file
+        ch_beam_antigen_panel_csv = params.beam_antigen_csv   ? file(params.beam_antigen_csv)   : empty_file
+        ch_frna_sample_csv        = params.frna_sample_csv    ? file(params.frna_sample_csv)    : empty_file
+        ch_cmo_barcode_csv        = params.cmo_barcode_csv    ? file(params.cmo_barcode_csv)    : empty_file
```

Comment:
That is my reasoning for bringing up to discussions. Because I am not very close to this data type, I want to understand whether having one per samplesheet would be ok, or if it would need to implement in a way that such file would go in the samplesheet, allowing one per included dataset.

Reply:
ok, I think the following would be most convenient for the user: 
 * provide a samplesheet as input with the physical samples + fastqs (as usual). The column `sample` corresponds to the "pysical sample" that got sequenced. e.g 
   ```
   sample,fastq_1,fastq_2
   SAMP001,/path/SAMP001_S1_L001_R1_001.fastq.gz,/path/SAMP001_S1_L001_R2_001.fastq.gz
   SAMP002,/path/SAMP001_S1_L001_R1_001.fastq.gz,/path/SAMP001_S1_L001_R2_001.fastq.gz
   ```
 * provide a second samplesheet that defines the multiplexed "sub samples". IMO it needs 4 columns:
   ```
   sample,multiplexed_sample_id,probe_barcode_ids,description
   SAMP001,SAMP001_ctrl,BC301,Control
   SAMP001,SAMP001_trt,BC302,Treated
   SAMP002,SAMP002_ctrl,BC301,Control
   SAMP002,SAMP002_trt,BC302,Treated
   ```

This samplesheet then needs to split up into
```
sample_id,probe_barcode_ids,description
SAMP001_ctrl,BC301,Control
SAMP001_ctrl,BC302,Control
```

and 

```
sample_id,probe_barcode_ids,description
SAMP002_ctrl,BC301,Control
SAMP002_ctrl,BC302,Control
```

and passed to the respective cellranger multi processes.